vapi/simulator/simulator.go

```diff
@@ -481,6 +493,47 @@ func (s *handler) category(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (s *handler) shutdown(w http.ResponseWriter, r *http.Request) {
+	delayShutdown := func(delay int) {
+		go time.AfterFunc(time.Duration(delay)*time.Minute, func() {
+			s.Lock()
+			s.Shutdown.ShutDownTime = ""
+			s.Shutdown.Action = ""
+			s.Shutdown.Reason = ""
+			s.Unlock()
+		})
+	}
+
+	switch r.Method {
+	case http.MethodGet:
+		OK(w, s.Shutdown)
+	case http.MethodPost:
+		switch r.URL.Query().Get("action") {
```

Comment:
This is the literal I meant where you could use the `const` approach too (as you do in the other `case` statements)

Reply:
changed to ```switch r.URL.Query().Get(shutdown.Action)```